ASM/c/file_icons.c

```diff
@@ -348,6 +386,12 @@ static void populate_fixed(const z64_file_t* file, fixed_tile_info_t* info) {
 
     // medallions
     info->medallion_bits = (uint8_t)(file->quest_items & 0x0000003F);
+    hasAllSixMeds = (info->medallion_bits & (1 << QUEST_MEDALLION_FOREST) &&
+                     info->medallion_bits & (1 << QUEST_MEDALLION_FIRE) &&
+                     info->medallion_bits & (1 << QUEST_MEDALLION_WATER) &&
+                     info->medallion_bits & (1 << QUEST_MEDALLION_SPIRIT) &&
+                     info->medallion_bits & (1 << QUEST_MEDALLION_SHADOW) &&
+                     info->medallion_bits & (1 << QUEST_MEDALLION_LIGHT));
```

Comment:
I'd take a look at how they did this for the ganon boss key condition.

Reply:
That is a lot more elegant than what i did, indeed. Thanks !